### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by anarang**
* Added setUpClass in test_rhai.py
* Added method for registering client to RHAI in 'rhai' module

### CODE REVIEW

File: robottelo/ui/rhai.py [hunk 1]
```diff
@@ -2,7 +2,13 @@
 """ Implements methods for RHAI"""
 
 from robottelo.ui.base import Base
```
Comment (elyezer) [inline robottelo/ui/rhai.py]: Please sort the import lines.

File: robottelo/ui/rhai.py [hunk 2]
```diff
@@ -12,3 +18,50 @@ def view_registered_systems(self):
...
+        with VirtualMachine(distro='rhel67') as vm:
+            # Download and Install ketello-ca rpm
+            vm.install_katello_cert()
+            vm.register_contenthost(activation_key, org)
+
+            # Red Hat Access Insights requires RHEL 6/7 repo and it not
+            # possible to sync the repo during the tests,
+            # adding a file in /etc/yum.repos.d/rhel6/7.repo
+
+            rhel6_repo = conf.properties.get('insights.rhel6_repo')
```
Comment (elyezer) [inline robottelo/ui/rhai.py]: If the config is not provided, the value of rhel6_repo will be None, should we raise an exception?

File: tests/foreman/rhai/test_rhai.py [hunk 3]
```diff
@@ -54,54 +46,37 @@ def test_rhai_client_setup(self):
...
-            self.assertEqual(result.return_code, 0)
+        @Assert: Registered client should appear in the Systems sub-menu of Red
+        Hat Access Insights
 
-            # Register client with Red Hat Access Insights
-            result = vm.run('redhat-access-insights --register')
-            self.assertEqual(result.return_code, 0)
+        """
+        # Register a VM to Access Insights Service
+        self.rhai.register_client_to_rhai(
```
Comment (elyezer) [inline tests/foreman/rhai/test_rhai.py]: The vm will be destroyed when this method returns and will not allow you to do any assertions. That makes sense?

File: tests/foreman/rhai/test_rhai.py [hunk 4]
```diff
@@ -112,7 +87,6 @@ def test_org_selection_for_rhai(self):
         org
 
         """
-
```
Comment (elyezer) [inline tests/foreman/rhai/test_rhai.py]: :

**Comment by elyezer** [comment]
ACK, pending comments.

File: tests/foreman/rhai/test_rhai.py (hunk 3 above)
Comment (anarang) [inline tests/foreman/rhai/test_rhai.py]: Yes, as the VM will get registered to the RHAI service and will appear on the UI and require to do assertion on the UI then, no matter if the VM is destroyed.

File: robottelo/ui/rhai.py (hunk 1 above)
Comment (anarang) [inline robottelo/ui/rhai.py]: Alright

File: robottelo/ui/rhai.py (hunk 2 above)
Comment (anarang) [inline robottelo/ui/rhai.py]: Agreed, added code for raising an exception.

**Comment by anarang** [comment]
Changes as per the comments done.

File: robottelo/ui/rhai.py (hunk 2 above)
Comment (Ichimonji10) [inline robottelo/ui/rhai.py]: Isn't it simpler to just do a dict lookup and let the lookup itself raise a `KeyError`?

```python
>>> from robottelo.common import conf
>>> conf.properties.get('foreman.admin.username')
'admin'
>>> conf.properties['foreman.admin.username']
'admin'
>>> conf.properties.get('foreman.admin.usernamee')
>>> conf.properties['foreman.admin.usernamee']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foreman.admin.usernamee'
```

**Comment by Ichimonji10** [comment]
Looks sane. ACK pending comment.

**Comment by anarang** [comment]
@Ichimonji10 @elyezer changes as per comments done.

**Comment by Ichimonji10** [comment]
@anarang I don't see your most recent changes on GitHub.

**Comment by Ichimonji10** [comment]
ACK. @elyezer, do you want to eyeball this?

**Comment by anarang** [comment]
@Ichimonji10 I missed one of the comments, updated now. Thanks :)

**Comment by Ichimonji10** [comment]
:+1:

**Comment by sthirugn** [comment]
ACK

**Comment by Ichimonji10** [comment]
I've reviewed all of @elyezer's comments, and all have been addressed. We have three ACKs. Merging.

**Pull request merged**